docs/source/methodology.md

```diff
@@ -0,0 +1,113 @@
+<!-- docs/source/methodology.md -->
+
+# Methodology
+
+PyDAGMC provides a Pythonic abstraction layer over PyMOAB for interacting with DAGMC (Direct Accelerated Geometry Monte Carlo) `.h5m` files. This section delves into the underlying concepts, data structures, and design choices that enable PyDAGMC's functionality.
```

Comment:
I wonder if we should include a statement about what a DAGMC file actually is?  Something like:

```
A DAGMC file contains a surface mesh representation of a geometry, including 
metadata to define the topological relationships of that geometry and material
assignments, using MOAB's mesh file conventions.  The metadata conventions are 
specific to DAGMC.
```

We do get into it more in the details below, but perhaps some early statement like this is helpful context and provides some useful vocabulary and semantic nuance.